### PR TITLE
reduce package `dependencies` cycles

### PR DESCRIPTION
**Pull request by turadg**
Refs: #4645
## Description

Burns down a cycle by moving some things to devDependencies that weren't runtime dependencies.

Burns down another by making Vats a parent of Inter Protocol. That required moving some token definitions _down_ to Inter Protocol (which is the purview) and moving some marshaling utils down to Internal (they depend on nothing in SDK).

Then it culls almost all of the `no-extraneous-dependencies` suppressions so we know that the dependencies declared in the package.jsons satisfy the actual imports.

After this there's just one left with chainStorage from /internal using `heapZone` from /zone. According to @erights and @michaelfig that package is destined for Endo. When that happens we can tighten the cycles check down to zero. (Note that checks only runtime dependencies and we'll still have devDependency cycles.)

### Testing Considerations

CI

### CODE REVIEW

File: packages/inter-protocol/src/proposals/addAssetToVault.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { AmountMath, AssetKind } from '@agoric/ertp';
 import { makeRatio } from '@agoric/zoe/src/contractSupport/index.js';
 import { deeplyFulfilledObject } from '@agoric/internal';
-import { Stable } from '@agoric/vats/src/tokens.js';
 import { E } from '@endo/far';
 import { parseRatio } from '@agoric/zoe/src/contractSupport/ratio.js';
+import { Stable } from '../tokens.js';
```
Comment (dckc) [inline packages/inter-protocol/src/proposals/addAssetToVault.js:8]: Interesting... I thought this (moving tokens to this package) would make zoe depend on inter-protocol, but now I remember... we pass the token name in as part of `feeIssuerConfig`... I think that's a result of the RUN->IST rename.

Consider moving the `feeIssuerConfig` from `packages/vats/src/core/utils.js` to `inter-protocol` as well.

**Comment by dckc** [comment]
I'm not sure how to review / evaluate this. It could be expensive to get me swapped in. Let's see... the issue says:

> This suggests a lack of overall architectural clarity.

Where do I look in this PR to get overall architectural clarity?

**Review by dckc — COMMENTED** [review]
As we discussed, I hope to land #7165, #7160 first.

File: packages/eslint-config/eslint-config.cjs [hunk 2]
```diff
@@ -77,8 +77,8 @@ module.exports = {
           '**/*.config.*.js',
           // leading wildcard to work in CLI (package path) and IDE (repo path)
           '**/test/**',
-          '**/demo*/**/*.js',
-          '**/scripts/**/*.js',
```
Comment (toliaqat) [inline packages/eslint-config/eslint-config.cjs:83]: Was this change intentional? There are other regexs above that are of old format.

Comment (turadg) [reply packages/eslint-config/eslint-config.cjs:83]: yes, to match `.mjs`: [ci(lint): glob mjs for devDependencies](https://github.com/Agoric/agoric-sdk/pull/7260/commits/b2a2d187698c809d457efa64994fa09e36a293b0)

The regex above don't need it though it wouldn't hurt to update

File: .github/workflows/test-all-packages.yml [hunk 3]
```diff
@@ -49,7 +49,7 @@ jobs:
         run: sudo apt install -y graphviz
 
       - name: Check for cycles
-        run: scripts/check-dependency-cycles.sh 3
+        run: scripts/check-dependency-cycles.sh 1
```
Comment (michaelfig) [inline .github/workflows/test-all-packages.yml:52]: 🙌

File: packages/SwingSet/package.json [hunk 4]
```diff
@@ -24,8 +24,7 @@
     "@types/better-sqlite3": "^7.5.0",
     "@types/microtime": "^2.1.0",
     "@types/tmp": "^0.2.0",
-    "better-sqlite3": "^8.2.0",
-    "yargs-parser": "^21.1.1"
+    "better-sqlite3": "^8.2.0"
```
Comment (michaelfig) [inline packages/SwingSet/package.json]: Why is `"better-sqlite3": "^8.2.0"` in both `devDependencies` and `dependencies`?

File: packages/solo/test/test-home.js [hunk 5]
```diff
@@ -5,9 +5,9 @@ import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
 import bundleSourceAmbient from '@endo/bundle-source';
 import { AmountMath } from '@agoric/ertp';
 import { TimeMath } from '@agoric/time';
-import { Stable } from '@agoric/vats/src/tokens.js';
 import { Far } from '@endo/marshal';
 import { resolve as importMetaResolve } from 'import-meta-resolve';
+import { Stable } from '@agoric/inter-protocol/src/tokens.js';
```
Comment (michaelfig) [inline packages/solo/test/test-home.js]: This seems suspect to me.  `ag-solo` should be independent of the economy, except for the fact that it may import the sim-chain from `@agoric/cosmic-swingset`.  Can you at least leave a comment behind describing how we would fix this layering?

One idea would be to move the part of the test relying on `Stable` to another package.

File: packages/solo/package.json [hunk 6]
```diff
@@ -26,6 +26,7 @@
     "@agoric/assert": "^0.6.0",
     "@agoric/cache": "^0.3.2",
     "@agoric/cosmic-swingset": "^0.41.3",
+    "@agoric/inter-protocol": "^0.16.1",
```
Comment (michaelfig) [inline packages/solo/package.json]: Can/should this be just a `devDependency`?

File: packages/xsnap-lockdown/package.json [hunk 7]
```diff
@@ -42,5 +42,8 @@
     ],
     "timeout": "2m",
     "workerThreads": false
+  },
+  "dependencies": {
```
Comment (michaelfig) [inline packages/xsnap-lockdown/package.json]: Why does taking a dependency on `@agoric/xsnap-lockdown` also require `@endo/init`?

File: packages/SwingSet/package.json (hunk 4 above)
Comment (turadg) [reply packages/SwingSet/package.json]: Thanks for catching. It was a devDependency for tests. Then `misc-tools` exported something that depends on it, so it needed to go in `dependencies`. 

I could imagine that there's a better place for those tools, but it wouldn't remove the dependency because it would still be included transitively by `@agoric/swingset-store`. So I'll just remove the redundant one in devDependencies.

File: packages/solo/test/test-home.js (hunk 5 above)
Comment (turadg) [reply packages/solo/test/test-home.js]: Glad to hear. This test was the only thing in `@agoric/solo` depending on inter-protocol. Since it's a test I've copied the config over. If the config changes the test should fail.

File: packages/solo/package.json (hunk 6 above)
Comment (turadg) [reply packages/solo/package.json]: Removed per ^^

File: packages/xsnap-lockdown/package.json (hunk 7 above)
Comment (turadg) [reply packages/xsnap-lockdown/package.json]: See c3b38d446a3ba98de2d88330ec0d5ea9ca97df15

Alternately, if the modules depending on Endo aren't meant to be exported, we can move them under `scripts` so they don't end up in the NPM distribution. cc @warner

File: packages/agoric-cli/src/chain-config.js
Comment (turadg) [inline packages/agoric-cli/src/chain-config.js:1]: @dckc PTAL

File: packages/xsnap-lockdown/package.json (hunk 7 above)
Comment (michaelfig) [reply packages/xsnap-lockdown/package.json]: The comment (removed in c3b38d4) explains the desired state of the world: the files in `lib` are only meant as build inputs for the generated bundles in `dist`.  All of `lib`'s dependencies should be only build-time dependencies, not runtime dependencies.

I'd suggest removing `"lib"` from `packageJson.files`, as well as moving `xsnap-lockdown/lib` to `xsnap-lockdown/scripts/lib` or somesuch (basically any appropriate directory that allows extraneous imports) and adjusting `scripts/build-bundle.js` accordingly.  Then I think it would make sense to put `"@endo/init"` back into `devDependencies`.

Comment (michaelfig) [reply packages/xsnap-lockdown/package.json]: Or even better, just remove `"lib"` from `packageJson.files`, remove `packageJson.dependencies`, and don't change the current package layout.

Comment (turadg) [reply packages/xsnap-lockdown/package.json]: I agree with the desired state. I removed the comment because we shouldn't suppress no-extraneous-dependencies.

I tried removing `lib` but `entryPaths` in `src/paths.js` depends on it and that does get exported. Removing no-extraneous-dependencies was some scope creep on this PR so I've reverted those changes to `xnap-lockdown`.

@michaelfig ready to approve? I'll cleanup the commits before merge.

File: packages/inter-protocol/src/proposals/addAssetToVault.js (hunk 1 above)
Comment (turadg) [reply packages/inter-protocol/src/proposals/addAssetToVault.js:8]: I looking into moving `feeIssuerConfig` but it's consumed only in `@agoric/vats` so moving wouldn't help the dependency graph.

File: packages/eslint-config/eslint-config.cjs (hunk 2 above)
Comment (erights) [reply packages/eslint-config/eslint-config.cjs:83]: Any concern that `*js` can have false hits like `Swarājs` ?

**Review by michaelfig — APPROVED** [review]
LGTM!  I'm willing to maintain the consequences. :)
